README.md

```diff
@@ -338,27 +338,108 @@ the [Applicative](#applicative) and [Plus](#plus) specifications.
 
 ### Foldable
 
-1. `u.reduce` is equivalent to `u.reduce((acc, x) => acc.concat([x]), []).reduce`
+1. `u` is equivalent to `u.foldl((acc, x) => acc.concat([x]), [])`
+2. `u.foldr(f, acc)` is equivalent to `u.foldl((a, b) => (c) => a(f(c,
+   b)), id)(acc)`
+3. `u.shortFoldl((acc, x) => {done: false, value: f(acc, x)},
+   initial)` is equivalent to `u.foldl(f, initial)`
+4. `u.shortFoldr((acc, x) => {done: false, value: f(x, acc)},
+   initial)` is equivalent to `u.foldr(f, initial)`
 
-#### `reduce` method
+#### `foldl` method
 
 ```hs
-reduce :: Foldable f => f a ~> ((b, a) -> b, b) -> b
+foldl :: Foldable f => f a ~> (b, a) -> b ~> b -> b
 ```
 
-A value which has a Foldable must provide a `reduce` method. The `reduce`
+A value which has a Foldable must provide a `foldl` method. The `foldl`
 method takes two arguments:
 
-    u.reduce(f, x)
+    u.foldl(f, x)
 
 1. `f` must be a binary function
 
-    1. if `f` is not a function, the behaviour of `reduce` is unspecified.
-    2. The first argument to `f` must be the same type as `x`.
+    1. if `f` is not a function, the behaviour of `foldl` is unspecified.
+    2. The _first_ argument to `f` must be the same type as `x`.
     3. `f` must return a value of the same type as `x`.
     4. No parts of `f`'s return value should be checked.
 
-1. `x` is the initial accumulator value for the reduction
+2. `x` is the initial accumulator value for the fold
+
+    1. No parts of `x` should be checked.
+
+#### `foldr`
+
+```hs
+foldr :: Foldable f => f a ~> (a, b) -> b ~> b -> b
+```
+
+A value which has a Foldable must provide a `foldr` method. The `foldr`
+method takes two arguments:
+
+    u.foldr(f, x)
+
+1. `f` must be a binary function
+
+    1. if `f` is not a function, the behaviour of `foldr` is unspecified.
+    2. The _second_ argument to `f` must be the same type as `x`.
+    3. `f` must return a value of the same type as `x`.
+    4. No parts of `f`'s return value should be checked.
+
+2. `x` is the initial accumulator value for the fold
+
+    1. No parts of `x` should be checked.
+
+#### `shortFoldl`
+
+```hs
+shortFoldl :: Foldable f => f a ~> ((b, a) -> {done :: Boolean, value :: b}) ~> b -> b
```

Comment:
Yes. I did indeed. I can see the appeal. But I choose an object for two reasons:

1. Array are more expensive to create. I.e. it would lower performance.
2. In Jabz I use an Either to signal when to stop. That is even nicer than a tuple IMO. But fortunately, with the current specification an Either implementation can include a `done` property which it is `true` for `left` and `false` for `right`. Then the specification will be directly compatible with that Either implementation. With that `find` could be implemented like this:

```ts
export function find<A>(f: (a: A) => boolean, t: Foldable<A>): Maybe<A> {
  return t.shortFoldl((acc, a) => f(a) ? left(just(a)) : right(acc), nothing);
}
```